src/test/components/elements/EventBookingForm.test.tsx

```diff
@@ -0,0 +1,193 @@
+import { AdditionalInformation, AugmentedEvent } from "../../../IsaacAppTypes";
+import { Immutable } from "immer";
+import { fireEvent, screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import { rest } from "msw";
+import * as actions from "../../../app/state/actions";
+
+const requestEmailVerificationSpy = jest.spyOn(actions, "requestEmailVerification");
+const updateAdditionalInformation = jest.fn();
+
+describe("EventBookingForm", () => {
+  const setupTest = ({
+    role,
+    event,
+    user,
+    additionalInformation,
+  }: {
+    role: UserRole;
+    event?: AugmentedEvent;
+    user: typeof mockUser;
+    additionalInformation?: AdditionalInformation;
+  }) => {
+    const mockUserSummary: Immutable<UserSummaryWithEmailAddressDTO> = {
+      givenName: user.givenName,
+      familyName: user.familyName,
+      role: role,
+      emailVerificationStatus: user.emailVerificationStatus as EmailVerificationStatus,
+      registeredContexts: user.registeredContexts as UserContext[],
+      id: user.id,
+      teacherPending: user.teacherPending,
+    };
+
+    renderTestEnvironment({
+      role: role,
+      PageComponent: EventBookingForm,
+      componentProps: {
+        event: event ? event : augmentEvent(mockEvent),
+        targetUser: mockUserSummary,
+        additionalInformation: additionalInformation ? additionalInformation : {},
+        updateAdditionalInformation: updateAdditionalInformation,
+      },
+      initialRouteEntries: [`/events/${mockEvent.id}`],
+      extraEndpoints: [
+        rest.post(API_PATH + `/users/verifyemail`, (req, res, ctx) => {
+          return res(ctx.status(200), ctx.json({}));
+        }),
+      ],
+    });
+  };
+
+  it("component renders", () => {
+    setupTest({ role: "STUDENT", user: mockUser });
+    const accountInformation = screen.getByTestId("booking-account-info");
+    const eventBookingDetails = screen.getByTestId("event-booking-details");
+    expect(accountInformation).toBeInTheDocument();
+    expect(eventBookingDetails).toBeInTheDocument();
+  });
+
+  it("has first name, last name , email address, context and school inputs, all disabled", async () => {
+    setupTest({ role: "STUDENT", user: mockUser });
+    await screen.findByDisplayValue(mockUser.givenName!);
+    const inputFields = [
+      "First name",
+      "Last name",
+      "Email address",
+      "Stage",
+      "Exam board",
+      "My current school or college",
+    ];
+    inputFields.forEach((input) => {
+      const inputField = screen.getByLabelText(input);
+      expect(inputField).toBeInTheDocument();
+      expect(inputField).toBeDisabled();
+    });
+  });
+
+  it("if email address is unverified, warning message is displayed", async () => {
+    const user = { ...mockUser, emailVerificationStatus: "NOT_VERIFIED" as EmailVerificationStatus };
+    setupTest({ role: "STUDENT", user: user });
+    await screen.findByDisplayValue(user.givenName!);
```

Comment:
You have used this line a few times in this test file, but is it really necessary? It looks redundant to me. For instance, in this test, shouldn't you simply use `getByTestId`? Or `findByTestId`?
I'm guessing that said line is there to wait until the content is loaded, but the same can be done with `findByTestId` (or whatever you actually need in each test).

Reply:
Yes it's used in any of the async tests to make sure the data has loaded. That's not necessary in all of them. Specifically I wanted to make sure that the "name" field contained the expected name, which indicates it has the user details.

This one actually doesn't need it so I've removed it entirely but I'll check the others to see if there's a more specific thing to await and adjust it. Does it make a functional difference however? Or just for clarity when reading the tests.

Reply:
> Does it make a functional difference however? Or just for clarity when reading the tests.

Just for clarity.